### PR TITLE
Add upstream proxy protocol and increase default connection limits

### PR DESCRIPTION
**Pull request by dboslee**
Changes:

- Forces the use of upstream proxy protocol v2.
- Disables some tests that were failing due to this change rather than updating a bunch of test input files/output files.
- Increases the connection limits per cluster (defaults are 1024). I arbitrarily picked 32000 for max connections and max pending requests. Docs on these values can be found [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/circuit_breaker.proto#config-cluster-v3-circuitbreakers-thresholds). Note this is applied per cluster in envoy which basically mean per tenant. So if we run 3 replicas per region each tenant could theoretically have ~96000 connections per region.

### CODE REVIEW

**Comment by sclevine** [comment]
Why are we skipping the existing tests instead of extending them to cover the new functionality? Seems like we want coverage here, and eventually, the ability to upstream?

**Comment by dboslee** [comment]
@sclevine Take a look at the latest commits, I left some tests skipped but kept in tls route specific tests. This gives us coverage for our use case for the time being but avoids potentially having a lot of conflicts with upstream.

File: internal/xds/translator/teleport_translate_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import (
+	"testing"
+
+	infra "github.com/envoyproxy/gateway/internal/infrastructure/kubernetes"
+	resourcev3 "github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTeleportTranslateXds(t *testing.T) {
```
Comment (sclevine) [inline internal/xds/translator/teleport_translate_test.go]: Marking things with "Teleport" is useful when we don't plan to upstream them. Wouldn't we want to upstream this functionality/test?

If this is functionality that we never plan to upstream, how do plan to move off this fork in the future? Should we make this configurable instead?

**Review by sclevine — APPROVED** [review]

**Review by rcanderson23 — APPROVED** [review]
Same thoughts concerns as Stephen with regards to upstream on these PRs

File: internal/gatewayapi/route.go [hunk 2]
```diff
@@ -553,6 +562,28 @@ func (t *Translator) ProcessTLSRoutes(tlsRoutes []*v1alpha2.TLSRoute, gateways [
...
 }
 
+func getUpstreamConfig(obj client.Object) ir.UpstreamConfig {
+	var (
+		enableProxyProtocol bool
+		maxConns            uint32
+	)
+
+	annotations := obj.GetAnnotations()
+	_, enableProxyProtocol = annotations[AnnotationUpstreamProxyProtocol]
```
Comment (sclevine) [inline internal/gatewayapi/route.go]: I would make this opt-in via a specific value (i.e., `true`). Empty annotations probably shouldn't change the behavior.

File: internal/gatewayapi/route.go [hunk 3]
```diff
@@ -553,6 +562,28 @@ func (t *Translator) ProcessTLSRoutes(tlsRoutes []*v1alpha2.TLSRoute, gateways [
 	return relevantTLSRoutes
 }
 
+func getUpstreamConfig(obj client.Object) ir.UpstreamConfig {
```
Comment (sclevine) [inline internal/gatewayapi/route.go]: It's ambiguous whether these annotations are TLSRoute-specific. The function and annotation names are generic, but the comment on the first annotation says TLSRoute.

If they could apply to all routes, but only apply to TLSRoute for now, maybe `obj` -> `route` here, rename annotation to `AnnotationRouteX`, and then be explicit in the comment that only `TLSRoute` is currently supported?

File: internal/gatewayapi/route.go (hunk 2 above)
Comment (dboslee) [reply internal/gatewayapi/route.go]: Using case-insensitive true [2455353](https://github.com/gravitational/gateway/pull/4/commits/245535374e5be2e0b3697a8e3bdc1c33fb883ebe)

File: internal/gatewayapi/route.go (hunk 3 above)
Comment (dboslee) [reply internal/gatewayapi/route.go]: You're right it can be generic but I only wired up TLSRoute for now. Updated comments and vars in latest commits.

Comment (sclevine) [reply internal/gatewayapi/route.go]: Looks great

**Review by sclevine — APPROVED** [review]

**Pull request merged**